@here/harp-mapview/lib/geometry/TileGeometryCreator.ts

```diff
@@ -711,55 +711,56 @@ export class TileGeometryCreator {
                 }
 
                 if (isLineTechnique(technique) || isSegmentsTechnique(technique)) {
-                    const hasDynamicColor =
-                        Expr.isExpr(technique.color) || Expr.isExpr(technique.opacity);
                     const fadingParams = this.getFadingParams(displayZoomLevel, technique);
                     FadingFeature.addRenderHelper(
                         object,
                         viewRanges,
                         fadingParams.fadeNear,
-                        fadingParams.fadeFar,
-                        false,
-                        hasDynamicColor
-                            ? (renderer, mat) => {
-                                  const lineMaterial = mat as THREE.LineBasicMaterial;
-                                  applyBaseColorToMaterial(
-                                      lineMaterial,
-                                      lineMaterial.color,
-                                      technique,
-                                      technique.color,
-                                      mapView.zoomLevel
-                                  );
-                              }
-                            : undefined
+                        fadingParams.fadeFar
                     );
+
+                    if (Expr.isExpr(technique.color) || Expr.isExpr(technique.opacity)) {
+                        tile.addUpdater(() => {
+                            const lineMaterial = material as THREE.LineBasicMaterial;
+                            applyBaseColorToMaterial(
+                                lineMaterial,
+                                lineMaterial.color,
+                                technique,
+                                technique.color,
+                                mapView.zoomLevel
+                            );
+                        });
+                    }
                 }
 
                 if (isSolidLineTechnique(technique)) {
-                    const hasDynamicColor =
-                        Expr.isExpr(technique.color) || Expr.isExpr(technique.opacity);
+                    const lineMaterial = material as SolidLineMaterial;
+
                     const fadingParams = this.getFadingParams(displayZoomLevel, technique);
                     FadingFeature.addRenderHelper(
```

Comment:
Can we also get rid of this render helper. It does not help anybody. Instead the material updated should also be part of an explicit method in Tile

Reply:
Some kind of helper is still needed, but it will be moved to TIleGeometryCreator and gratly simplified.